backend/production-optimiser-api/src/main/java/it/polimi/productionoptimiserapi/controllers/AccountRequestController.java

```diff
@@ -0,0 +1,58 @@
+package it.polimi.productionoptimiserapi.controllers;
+
+import it.polimi.productionoptimiserapi.dtos.AccountRequestDTO;
+import it.polimi.productionoptimiserapi.dtos.KeyValueDTO;
+import it.polimi.productionoptimiserapi.dtos.UserDTO;
+import it.polimi.productionoptimiserapi.services.AccountRequestService;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/account-requests")
+@RequiredArgsConstructor
+public class AccountRequestController {
+
+  private final AccountRequestService accountRequestService;
+
+  @GetMapping("/{id}")
+  @PreAuthorize("hasRole('ADMIN')")
+  public AccountRequestDTO getAccountRequest(@PathVariable String id) {
+    return accountRequestService.getAccountRequest(id);
+  }
+
+  @GetMapping("/all")
+  @PreAuthorize("hasRole('ADMIN')")
+  public List<AccountRequestDTO> getAccountRequests() {
+    return accountRequestService.getAccountRequests();
+  }
+
+  @PostMapping("/form")
```

Comment:
same as /all, PostMapping would be better

Reply:
yeah I referenced this in PR description: 
"sidenote:
had to deviate from normal endpoint urlformat for "createAccountRequest" because I couldn't find another way to bypass security"
I agree it should be without "/form" but when changing securityConfig and adding 
.requestMatchers(HttpMethod.POST, UrlConstants.ACCOUNT_REQUEST_URL).permitAll() I kept getting status Forbidden
if anyone has any suggestions absolutely but nothing I tried worked